### DISCUSSION TITLE
Use Entra ID Username instead of Effetive Username when connecting as a person on DAX Studio

### DISCUSSION THREAD

**Discussion by VincentGvr** [DaxStudio/DaxStudio#1211] [General]
Hi, 

I've been testing some queries on a dataset using Single Sign On to Snowflake with DAX Studio. In order to propagte the identity of the user to test RLS on the datasource, I'm using the Effective User Name from the Advanced Option of the Connect menu. 

The problem is that I have a "Premium_ASWL_Error", telling me that EffectiveUserName cannot be compatible with the SSO.

I'm wondering if there is a way to connect with a user like username@contoso.com, so Entra Id username from DAX Studio, and still avoid using the Effective User Name ? In other words, can I use another way to connect that the Effective User Name if I want to test with another user than mine ?

**Comment by dgosbell** [comment]
This is not really something the I can solve with DAX Studio. This is more to do with how SSO works. Basically, when you authenticate with SSO there is a single auth token that gets pass from the client (DAX Studio) to the dataset and then on to Snowflake. This token is digitally signed and cannot be altered. This is generally a good thing as you don't want people being able to impersonate someone else and access data they should not have access to.

You don't run into this issue with Import mode models or models with fixed credentials - since the impersonated user does not need to be passed on to another system.

**Reply by VincentGvr** [reply]
Hi, thank you very much for your quick answer. 

In our case, data is stored in Snowflake. We are doing a benchmark to compare : 
- a model in Import Mode with RLS in Power BI, with the challenge of fitting all data from snowflake in aggregated way, 
- a hybrid model in import mode + facts in Direct Query and RLS in Power BI 
- a model fully in DQ with RLS at the source level, using Single Sign On. 

To obtain the full level of details we need, we were comparing the same visuals using the 3 techniques, with Query times and engine times, but we faced the issue with SSO & Snowflake. I'm fully aligned with the fact that Power BI engines runs In Memory and that we should avoid going with DQ, but in order to explain this to the team, I need proofs. and the proofs reside in the benchmark. 

Can I say that DAX Studio does not provied a way to test SSO w/ classic authentication and without EffectiveUserName, as we can do with the "View As Role" in Power BI Desktop or on the Service ?

**Reply by dgosbell** [reply]
I'm not sure what you mean by "classic authentication" that's not a term I've heard before. You should be able to test SSO using DAX Studio. You just have to authenticate as the user you want to test when you first connect to the semantic model. You could then open a second instance of DAX Studio and sign in as a different user and you should be able to observe the SSO working with the RLS at the source. 

Neither the EffectiveUserName or the View As option in DAX Studio nor the View As Role in Desktop or the Service are compatible with SSO. 

It is fundamental to the  whole concept of SSO that only a single identity is used.
